spec/jsonapi/authorization/default_pundit_authorizer_spec.rb

```diff
@@ -213,55 +213,66 @@
+      before { stub_policy_actions(source_class, create_with_comments?: true, create?: true) }
+      it { is_expected.not_to raise_error }
+    end
 
-      context 'unauthorized for update? on any of the related records' do
-        let(:related_records) { [Comment.new(id: 1), Comment.new(id: 2)] }
-        before do
-          allow_action(related_records.first, 'update?')
-          disallow_action(related_records.last, 'update?')
-        end
+    context 'authorized for create? and unauthorized for create_with_<type>? on source class' do
+      let(:related_records) { [Comment.new(id: 1), Comment.new(id: 2)] }
+      before { stub_policy_actions(source_class, create_with_comments?: false, create?: true) }
+      it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
+    end
 
-        it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
-      end
+    context 'unauthorized for create? on source class and related records is empty' do
+      let(:related_records) { [] }
+      before { stub_policy_actions(source_class, create?: true) }
+      it { is_expected.not_to raise_error(::Pundit::NotAuthorizedError) }
     end
 
-    context 'unauthorized for create? on source class' do
-      before { disallow_action(source_class, 'create?') }
+    context 'unauthorized for create? and authorized for create_with_comments? on source class' do
+      before { stub_policy_actions(source_class, create_with_comments?: true, create?: false) }
+      it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
+    end
 
-      context 'related records is empty' do
-        let(:related_records) { [] }
-        it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
-      end
+    context 'unauthorized for create? and unauthorized for create_with_comments? on source class' do
+      let(:related_records) { [Comment.new(id: 1), Comment.new(id: 2)] }
+      before { stub_policy_actions(source_class, create_with_comments?: false, create?: false) }
+      it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
+    end
 
-      context 'authorized for update? on all of the related records' do
-        before { related_records.each { |r| allow_action(r, 'update?') } }
-        it { is_expected.to raise_error(::Pundit::NotAuthorizedError) }
+    context 'authorized for create? where create_with_<type>? is undefined' do
+      context 'authorized for update? on related records' do
+        before do
+          stub_policy_actions(source_class, create?: true)
+          related_records.each { |r| stub_policy_actions(r, update?: true) }
+        end
+        it { is_expected.not_to raise_error(::Pundit::NotAuthorizedError) }
       end
 
-      context 'unauthorized for update? on any of the related records' do
-        let(:related_records) { [Comment.new(id: 1), Comment.new(id: 2)] }
+      context 'unauthorized for update? on related records' do
```

Comment:
This might be a bit of a nitpick, but do you think it'd be a good idea to test this as

```
context 'unauthorized for update? on any of the related records'
```

...and then just stub one of the `related_records` to return `false` for `update?` and stub others to return `true`? It would be similar to what there used to be.